src/hotspot/share/opto/loopopts.cpp

```diff
@@ -1272,9 +1272,7 @@ Node* PhaseIdealLoop::place_outside_loop(Node* useblock, IdealLoopTree* loop) co
   // Pick control right outside the loop
   for (;;) {
     Node* dom = idom(useblock);
-    if (loop->is_member(get_loop(dom)) ||
-        // NeverBranch nodes are not assigned to the loop when constructed
-        (dom->is_NeverBranch() && loop->is_member(get_loop(dom->in(0))))) {
+    if (loop->is_member(get_loop(dom))) {
```

Comment:
This would not be backported, right? Do you think we should do it in a separate RFE? Or is it necessary for the fix?

Reply:
Yes, it should be separate RFE  or you back port [JDK-8335709](https://bugs.openjdk.org/browse/JDK-8335709) first.

Reply:
Right. That makes sense. I removed it in latest commit and will create a separate PR for it.